test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesWithBasePathTest.cs

```diff
@@ -391,5 +392,70 @@ public RazorPagesWithBasePathTest(MvcTestFixture<RazorPagesWebSite.StartupWithBa
             // Assert
             Assert.Equal("Hello from AllowAnonymous", response.Trim());
         }
+
+        [Fact]
+        public async Task PageConventions_CanBeUsedToCustomizeTheModelType()
```

Comment:
I think it's worth a comment here and around unit tests, calling out that this kind of test is important because you need this to work smoothly when a page is using a base class as its declared type to work around some limitations we have. I think it's important to have a comment mapping back to a users scenario. 

Someone like me might otherwise say "this is a dumb tests, let's remove it" or "this is a dumb feature, let's try to take it out".

Reply:
I'll add a comment